src/useCase/convertTemperature/ConvertTemperatureUseCase.ts

```diff
@@ -0,0 +1,44 @@
+// CELSIUS > KELVIN = CELSIUS + 273
+// CELSIUS > FARENHEIT = 1.8 * CELSIUS + 32
+
+// KELVIN > CELSIUS = KELVIN - 273
+// KELVIN > FARENHEIT = (converter kelvin para celsius e realizar CELSIUS > FARENHEIT)
+
+// FARENHEIT > CELSIUS = (FARENHEIT - 32) / 1.8
+// FARENHEIT > KELVIN = (converter farenheit para celsius e realizar CELSIUS > KELVIN)
+
+interface IData {
```

Comment:
Gu, você consegue tipar o valor da propriedade e scale e scaleToConvert.
![image](https://user-images.githubusercontent.com/98772000/213717034-832f546d-6774-4977-9078-c2e312ba8fe6.png)
Neste caso, a pessoa só vai conseguir passar "fahreinheit", "celsius" ou "kelvin" na propriedade scale e scaleToConvert. Assim, meio que quando você passar um outro valor ao chamar a função, ele vai reclamar falando que não existe na tipagem dele, todavia no Front-end e no Insomnia é possível passar um valor diferente e quebrar, sendo assim, necessário uma validação dentro da Classe.

Você pode usar o enum quanto o type. Entretando os dois são diferente e usados em casos. 

Vou deixar 2 links bem bacanas para você dar uma olhada:
- Documentação
https://www.typescriptlang.org/docs/handbook/enums.html

- Os perigos do Enum
https://dev.to/oieduardorabelo/os-perigos-dos-enums-em-typescript-47g1